crates/jarscanner/src/lib.rs

```diff
@@ -99,10 +131,11 @@ pub fn process_input(
 pub fn emit_result(
     target_descriptor: &ExtractedData,
     output_path: &PathBuf,
-) -> Result<(), Box<dyn Error>> {
+) -> Result<(), JarscannerError> {
     let json = serde_json::to_string_pretty(target_descriptor)?;
```

Comment:
https://docs.rs/serde_json/latest/serde_json/fn.to_writer_pretty.html

how about instead don't build the `json`:
```rust
pub fn emit_result(
    target_descriptor: &ExtractedData,
    output_path: &PathBuf,
) -> Result<(), JarscannerError> {
    let file = File::create(output_path)?;
    let writer = BufWriter::new(file);
    serde_json::to_writer_pretty(writer, target_descriptor)?;
    Ok(())
 }
```
this may skip a copy of the final output which can also be a performance win.